### DISCUSSION TITLE
Just confirming: is `react-compiler-runtime` a prod or dev dependency?

### DISCUSSION THREAD

**Discussion by TowhidKashem** [reactwg/react-compiler#41] [General Questions] [answered]
I think I know the answer given the name but normally we associate react compiler with build time activities which are usually dev dependencies. But with "runtime" in the name, this one sounds like it's needed in prod. Can someone please confirm if this is the case?

**Comment by faradaytrs** [comment, marked as answer]
yes react-compiler-runtime should be available in runtime, this package is only required for react builds lower then 19, react 19 build already includes all necessary runtime code.

**Comment by guillaumebrunerie** [comment]
It's not particularly surprising for a compiler to have a runtime element. Many compiled languages have a runtime GC for instance. In the case of the React compiler, I guess it is mainly the `_c` function that you can see being used in the playground for memoization.
